workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/InformLicensingTermsPage/InformLicensingTermsPage.tsx

```diff
@@ -142,29 +165,87 @@ export function InformLicensingTermsPage({
 								});
 							}
 
-							const skuJSON = await getSKUById(skuId);
+							const skuJSON = await getSKUById(skuVersionId);
 
 							const skuBody = {
 								...skuJSON,
 								neverExpire: appLicense === 'perpetual',
 								price: 0,
 								purchasable: true,
+								skuOptions: [
+									{
+										key: productOptionId,
+										value:
+											dayTrial === 'yes'
+												? optionValuesId.yesOptionId
+												: optionValuesId.noOptionId,
+									},
+								],
 							};
 
-							await patchSKUById(skuId, skuBody);
+							await patchSKUById(skuVersionId, skuBody);
 						}
 
 						if (dayTrial === 'yes' && priceModel !== 'free') {
-							createAppLicensePrice({
+							const skuResponse = await getProductSKU({
 								appProductId,
-								body: {
-									neverExpire: true,
-									price: 0,
-									published: true,
-									purchasable: true,
-									sku: 'trial',
-								},
 							});
+
+							const trialSku = skuResponse.items.find(
+								({sku}) =>
+									sku ===
+									createSkuName(
+										appProductId,
+										appVersion,
+										'ts'
+									)
+							);
+
+							let skuTrialId;
+
+							if (trialSku) {
+								skuTrialId = trialSku.id;
+							}
+							else {
```

Comment:
Hey @naraandrade, I couldn't create a `ProductSubscriptionConfiguration` here, because I realized that we don't have an API to post a `productSubscriptionConfiguration`. Just Patch, and it doesn't get an array. So I would end up upgrading the old subscription.